### ISSUE TITLE
Fault resilience: revert behaviour?

### ISSUE THREAD

**Issue by hydro-b** [Scalingo/link#88]
Maybe I interpret the following wrong:

** Fault resilience** If for any reason something went wrong (lost connection with etcd) LinK will always try to have at least one host this means that if one agent fails to contact the etcd cluster it will take the IP.

Does the above mean that the host that *fails* to contact the etcd cluster will take the ip? And that another host might also take the ip?

Is it possible to configure link in such a way that at *most* one host takes the ip? In case a node loses connectivity to the etcd cluster something is wrong with it and it should drop all it's responsibilities, instead of taking the responsibility. Only healthy nodes (that can contact the etcd cluster) should be allowed to take an ip.

**Comment by johnsudaar** [comment]
Hi !

Thanks a lot for your input. I love your idea. Different faut resillience modes could be a nice addition to LinK. How would you see this feature ? As a general configuration for LinK or something that should be configurable per IP ?

**Comment by hydro-b** [comment]
A general configuration option would be good enough (for us anyway). I doubt you want to mix behaviour in one setup, although this would mean ultimate flexibility. Not sure what would be easier to implement though.

TL;DR: In scenarios were consistency is more important than availability.
We would use it to keep at most 1 IP available on a (HA-) proxy node. The drawback of VRRP / CARP (and the potential beauty of LinK) is that in a split-brain scenario the node will take the ip, and you might end up with duplicate IPs and that might screw things up.

**Comment by matthieu526-scalingo** [comment]
Closing this issue is more than 3 years old, and likely won't be tackled as is. It can always be reopened.
